corehq/apps/user_importer/importer.py

```diff
@@ -582,35 +579,35 @@ def create_or_update_commcare_users_and_groups(upload_domain, user_specs, upload
                     # Passing use_primary_db=True because of https://dimagi-dev.atlassian.net/browse/ICDS-465
                     user.get_django_user(use_primary_db=True).check_password(password)
 
-                for group in domain_info.group_memoizer.by_user_id(user.user_id):
-                    if group.name not in group_names:
-                        group.remove_user(user)
+                error = commcare_user_importer.update_user_groups(domain_info, group_names)
+                if error:
+                    error_message = f"{user.user_id}: {error}"
```

Comment:
I don't think you need to add user_id to the message, the error message will already be bound to the user row in `ret` no?
Plus, if we need to add a user identity here, we should use username so that when the users see it, they can link it to a user.

Reply:
Yeah, that makes sense. I'll take out the user_id.